### PR TITLE
Large update to FE build/dist

### PR DESCRIPTION
**Pull request by gmjosack**
We're now using npm to manage our frontend dev dependencies and
gulp to manage our front end builds

* Add some node files and built assets to .gitignore
* Gulp added w/ tasks for linting, caching templates, annotating ng DI,
  concat, minify, etc.
* Setup npm devDependencies and shrinkwrap them for consistent build
* Relocated js/css into src directory that isn't included with dist build
* Updated angular code to not explicitly put DI params twice since that
  happens at build
* Angular templates are now compiled to javascript and added to the
  template cache
* Fixed some lint errors (semicolons!)
* setup.py updated to support running all tests (python & javascript)
* setup.py updated to build static on develop/sdist commands
* Removed 3rd party deps from the checked in repo
* Fixed MANIFEST.in to not include pyc's under tests

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+ */
+gulp.task('lint:js', function() {
+    return gulp.src(JS_SRC)
+        .pipe(jshint())
+        .pipe(jshint.reporter('jshint-stylish'));
+});
+
+
+/**
+ * Task to CSS files JavaScript files.
```
Comment (jathanism) [inline gulpfile.js]: Wat?

File: setup.py [hunk 2]
```diff
@@ -18,10 +29,62 @@ def get_package_data(package, base_dir):
...
+
+
+class SDistWithBuildStatic(SDistCommand):
+    def make_distribution(self):
+        self.run_command('build_static')
+        return SDistCommand.make_distribution(self)
+
+
+class BuildStatic(Command):
+    def initialize_options(self):
```
Comment (jathanism) [inline setup.py]: Can you please add `user_options = []` so this can be run individually from the CLI?

File: gulpfile.js (hunk 1 above)
Comment (gmjosack) [inline gulpfile.js]: Fixed

File: setup.py (hunk 2 above)
Comment (gmjosack) [inline setup.py]: Done

**Comment by jathanism** [comment]
Also this fixes #60

**Pull request merged**